src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -29,66 +30,116 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
-            //
-            // Note: we only check the top level statements of the block.  I think that's sufficient
-            // as this will catch the 90% case, while not being that bad an experience even when 
-            // people do strange things in their constructors.
-            if (blockStatementOpt != null)
+            foreach (var parameterNode in listOfParameterNodes)
             {
-                if (!CanOffer(blockStatementOpt.Syntax))
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
+                if (ParameterValidForNullCheck(document, parameter, semanticModel, blockStatementOpt, cancellationToken))
                 {
-                    return ImmutableArray<CodeAction>.Empty;
+                    listOfParametersOrdinals.Add(parameter.Ordinal);
                 }
+            }
 
-                foreach (var statement in blockStatementOpt.Operations)
+            if (!listOfParametersOrdinals.Any())
+            {
+                return ImmutableArray<CodeAction>.Empty;
+            }
+
+            // ***  Could have a check here if the list only has one parameter and it is the one pressed - then do not offer add null check for all *** \\ 
+
+            // Great.  The list has parameters that need null checks. Offer to add null checks for all.
+
+            var result = ArrayBuilder<CodeAction>.GetInstance();
+
+            result.Add(new MyCodeAction(
+                    $"Add null checks for all",
+                    c => DocumentRetriver(document, functionDeclaration, methodSymbol, blockStatementOpt, listOfParametersOrdinals, c)));
+
+            return result.ToImmutableAndFree();
+
+
+            // DocumentRetriver updates all necessary variables per each parameter and calls for AddNullCheckAsync
+            async Task<Document> DocumentRetriver(
+                Document document, SyntaxNode functionDeclaration, IMethodSymbol methodSymbol, IBlockOperation blockStatementOpt, List<int> listOfParametersOrdinals, CancellationToken cancellationToken)
+            {
+                foreach (var index in listOfParametersOrdinals)
                 {
-                    if (IsIfNullCheck(statement, parameter))
+                    // Update functionDeclaration and use it to get the first valid ParameterNode using the ordinals (index).
+                    var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var token = root.FindToken(position);
+                    var firstparameterNode = GetParameterNode(token, position);
+
+                    functionDeclaration = firstparameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
```

Comment:
I think using syntax annotation to keep track of the method declaration node would make this simpler and less error prone.

Reply:
why was this resolved?